packages/network-controller/src/NetworkController.ts

```diff
@@ -1231,6 +1231,25 @@ export class NetworkController extends BaseControllerV2<
     });
   }
 
+  /**
+   * Searches for a network configuration ID with the given ChainID and returns it.
+   *
+   * @param chainId - ChainId to search for
+   * @returns networkClientId of the network configuration with the given chainId
+   */
+  findNetworkClientIdByChainId(chainId: Hex): NetworkClientId {
+    const networkClients = this.getNetworkClientRegistry();
+    const networkConfigurationIndex = Object.values(networkClients).findIndex(
+      (networkClient) => {
+        return networkClient.configuration.chainId === chainId;
+      },
+    );
+    if (networkConfigurationIndex === -1) {
+      throw new Error("Couldn't find networkClientId for chainId");
+    }
+    return Object.keys(networkClients)[networkConfigurationIndex];
+  }
+
```

Comment:
does Object.values and Object.keys guarantee elements ordered the same? Would using `Object.entries()` be worth it?